.github/scripts/pr-triage.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+set -euo pipefail
+
+# Initialize a comma-separated string to hold PR numbers that need a comment
+PRS_NEEDING_COMMENT=""
+
+# Function to process a single PR
+process_pr() {
+    local PR_NUMBER=$1
+    echo "🔄 Processing PR #$PR_NUMBER"
+
+    # Get PR body with error handling
+    local PR_BODY
+    if ! PR_BODY=$(gh pr view "$PR_NUMBER" --repo "$GITHUB_REPOSITORY" --json body -q .body 2>/dev/null); then
+        echo "   ⚠️ Could not fetch PR #$PR_NUMBER details"
+        return 1
+    fi
+
+    # Look for issue references using multiple patterns
+    local ISSUE_NUMBER=""
+
+    # Pattern 1: Direct reference like #123
+    if [ -z "$ISSUE_NUMBER" ]; then
+        ISSUE_NUMBER=$(echo "$PR_BODY" | grep -oE '#[0-9]+' | head -1 | sed 's/#//' 2>/dev/null || echo "")
+    fi
+
+    # Pattern 2: Closes/Fixes/Resolves patterns (case insensitive)
+    if [ -z "$ISSUE_NUMBER" ]; then
+        ISSUE_NUMBER=$(echo "$PR_BODY" | grep -iE '(closes?|fixes?|resolves?) #[0-9]+' | grep -oE '#[0-9]+' | head -1 | sed 's/#//' 2>/dev/null || echo "")
+    fi
```

Comment:
Instead of grepping the issue text, linked issues could be retrieved from the API via a `gh ...` query or the JS API.